src/common.ts

```diff
@@ -59,8 +59,12 @@ export const createAtoms = <
     const observable = {
       subscribe: (arg: { next: (result: Result) => void }) => {
         const callback = (result: Result) => {
-          if (!(observer as any)[SKIP_LISTENERS]) {
-            ;(typeof arg === 'function' ? arg : arg.next)(result)
+          const notifyResult = () =>
+            (typeof arg === 'function' ? arg : arg.next)(result)
+          if ((observer as any)[IN_RENDER]) {
+            Promise.resolve().then(notifyResult)
```

Comment:
Just curious how this works. Isn't this just the same as `notifyResult()`?

Reply:
It delays for one tick or so. This is a hack to avoid React warning. I wish there's a better solution.

Reply:
Interesting. I usually use `setTimeout(...,0)` in such cases and I thought resolved promise will continue immediately avoiding queuing but looks like it doesn't although it does have a higher priority than `setTimeout`.

```
(() => {
  setTimeout(()=>console.log(1), 0);
  Promise.resolve().then(() => console.log(2));
  console.log(3);
})()
```
```
3
2
1
```